### PR TITLE
DBZ-137 log miner oracle contribution

### PR DESCRIPTION
**Pull request by AndreyIg**

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+     * @param connection connection
+     * @return size
+     */
+    private static int getRedoLogGroupSize(Connection connection) throws SQLException {
+        return getMap(connection, SqlUtils.ALL_ONLINE_LOGS, "-1").size();
+    }
+
+    /**
+     * This method returns all online log files, starting from one which contains offset SCN and ending with one containing largest SCN
+     * 18446744073709551615 on Ora 19c is the max value of the nextScn in the current redo todo replace all Long with BigDecimal for SCN
```
Comment (MilovdZee) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:369]: BigInteger. Is not a floating point.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+        Map<String, String> redoLogFiles = getMap(connection, SqlUtils.ALL_ONLINE_LOGS, "-1");
+        return redoLogFiles.entrySet().stream().
+                filter(entry -> new BigDecimal(entry.getValue()).longValue() > offsetScn || new BigDecimal(entry.getValue()).longValue() == -1).
+                collect(Collectors.toMap(Map.Entry::getKey, e -> new BigDecimal(e.getValue()).longValue() == -1 ? Long.MAX_VALUE : new BigDecimal(e.getValue()).longValue()));
+    }
+
+    private static void executeCallableStatement(Connection connection, String statement) throws SQLException {
+        Objects.requireNonNull(statement);
+        CallableStatement s;
+        s = connection.prepareCall(statement);
```
Comment (MilovdZee) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:381]: use try with closable resource

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+     * @throws SQLException if anything unexpected happens
+     */
+    static void checkSupplementalLogging(OracleConnection connection, String pdbName) throws SQLException {
+        try {
+            if (pdbName != null) {
+                connection.setSessionToPdb(pdbName);
+            }
+
+            final String key = "KEY";
+            String validateGlobalLogging = "SELECT '" + key + "', " + " SUPPLEMENTAL_LOG_DATA_ALL from V$DATABASE";
```
Comment (MilovdZee) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:265]: cleanup

**Comment by MilovdZee** [comment]
I'm checking the functionality in my own code and find that commits are missed. A commit is done on the SCN of the original insert/update and if we are already past that SCN then the commit is not seen. How is that working in this code because as I see it this has the same issue...

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 3 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:265]: I don't see a problem with this as the variable is used later in an equality check.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 2 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:381]: Applied in PR #185.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 1 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:369]: Applied in PR #185.

**Comment by Naros** [comment]
This PR is superseded by PR #185.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 3 above)
Comment (MilovdZee) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:265]: I mean all the string concattenation on one line. No biggy but I just noticed it.

**Pull request closed without merging**